lib/awful/screen.lua

```diff
@@ -505,6 +564,100 @@ function screen.object.get_selected_tag(s)
     return screen.object.get_selected_tags(s)[1]
 end
 
+local function normalize(ratios, size)
+    local sum = 0
+
+    for _, r in ipairs(ratios) do
+        sum = sum + r
+    end
+
+    -- Avoid to mutate the input.
+    local ret = {}
+    local sum2 = 0
+
+    for k, r in ipairs(ratios) do
+        ret[k] = (r*100)/sum
+        ret[k] = math.floor(size*ret[k]*0.01)
+        sum2 = sum2 + ret[k]
+    end
+
+    -- Fix the sub-pixel issue.
+    ret[#ret] = ret[#ret] + (size - sum2)
```

Comment:
I would prefer something smarter than just giving "everything that was rounded away" to the last item, but since I do not know yet what this function is used for, I do not know whether this matters much.

Reply:
The main use case for this, as documented somewhere (I think it is in another unsubmitted PR), is that ultrawide monitors are larger than the field of view.

Lets assume that the rational from moving from the classic 4:3 ratio was to align the screen ratio to the common people field of view and that they done their homework.

Lets assume that any window that's larger than the user field of view is useless. Yes, some apps have their own tiling like blender or use canvas and tools like sidebars. Let's ignore those. Nobody wants a web browser spanning a 32:9 ratio with an hardcoded 1024 pixel large blog with aligned in the middle.

Now, screens with 21:9 and 32:9 ratios are being sold. I have one.

They are totally unusable with the current tiling system because clients get too large. That leaves either using custom layouts and/or configuration like @actionless `master_fill_policy` or split the screens into many virtual screens.

Virtual screens are superior because some common layouts like `awful.layout.suit.max` don't have `master_fill_policy`. Virtual screens also allow different taglists. This emulates the classic dual/tri-screen workflow, but without the gaps between the physical screens.